codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -48,8 +49,16 @@
  * ...
  * p.addLast("handler", new HttpRequestHandler());
  * </pre>
- * Be aware that you need to have the {@link HttpResponseEncoder} or {@link HttpRequestEncoder}
- * before the {@link HttpObjectAggregator} in the {@link ChannelPipeline}.
+ * Be aware that you need to have the {@link HttpResponseDecoder} or {@link HttpRequestDecoder}
+ * before the {@link HttpObjectAggregator} in the {@link ChannelPipeline} for it to receive
+ * {@link HttpObject}s to aggregate, and you need a {@link HttpResponseEncoder} before the
+ * {@link HttpObjectAggregator} if it will be receiving {@link HttpRequest} messages as the
+ * {@link HttpObjectAggregator} may write a 100 Continue response.
```

Comment:
How about: 

```
 , and a {@link HttpResponseEncoder} needs to be before the
 {@link HttpObjectAggregator} if it will be receiving {@link HttpRequest} messages as the
 {@link HttpObjectAggregator} is responsible for responding to a
 '100-continue' expectation or when content exceeds maxContentLength.
```